### DISCUSSION TITLE
Missing hover info in C++ - configuration or client issue?

### DISCUSSION THREAD

**Discussion by shahamran** [zed-industries/zed#20010] [Feature Requests]
Hi, I'm asking here since I don't know if it's a misconfiguration on my side, or a missing feature of Zed as an LSP client.

When editing C++ in VSCode with clangd as language server, hover box shows very useful information that isn't displayed in Zed, like:
* Enum variant discriminant value: 
* Size and offset of each struct field
* Size and alignment of entire struct

Consider the following snippet:
```cpp
#include <cstdint>

enum Enum : std::uint8_t {
    First = 0, // first
    // important thing!
    Second,
};

struct Struct {
    std::uint8_t small;
    std::uint32_t medium;
};
```

VSCode hovers:
<img width="313" alt="image" src="https://github.com/user-attachments/assets/698ec271-8472-4534-8665-1862de4356fa">
<img width="551" alt="image" src="https://github.com/user-attachments/assets/2897a00d-5a30-439d-90bd-e481ad297cc6">
<img width="352" alt="image" src="https://github.com/user-attachments/assets/622e2ed8-c0c2-4d8b-bc66-f24fe3af49a6">

Zed hovers:
<img width="370" alt="image" src="https://github.com/user-attachments/assets/2cc13c70-276f-4a7c-a328-1b947bda49a7">
<img width="492" alt="image" src="https://github.com/user-attachments/assets/efaacdc3-5ced-4726-90c6-5a1ee21a3623">
<img width="361" alt="image" src="https://github.com/user-attachments/assets/e520e559-3b16-43f6-bb27-16d5aec87337">

I use the default clangd configuration in both editors as far as I can tell. Any idea?

Thanks in advance.

**Comment by SomeoneToIgnore** [comment]
This is a regression, unfortunately, tracked here: https://github.com/zed-industries/zed/issues/15498

**Reply by shahamran** [reply]
Thanks @SomeoneToIgnore. I see it's a pretty long-standing issue. Is it on any roadmap? This feature is essential in my $dayjob.

**Reply by SomeoneToIgnore** [reply]
We don't really have any roadmaps for such things, someone has to go and debug it.

I am fighting the outline panel perf issues this week, so definitely not able to fix it soon-ish, alas.

**Reply by shahamran** [reply]
Do you think it's something I can try to tackle myself? or does it require too much specific knowledge on the editor's internals?

Didn't contribute to Zed before, but I'm OK in Rust and debugging in general :)

**Reply by SomeoneToIgnore** [reply]
I think it's pretty external, so welcome.

The starting point is somewhere around https://github.com/zed-industries/zed/blob/633b665379c18a06959102c3857889b8bcbde4bb/crates/project/src/project.rs#L2671-L2672 and you need to check two things:
* does the actual LSP response send you what's needed? (you can do that without the code, and seems that we do get all the data according to https://github.com/zed-industries/zed/issues/15498#issuecomment-2393413492 and the message below)
* do we parse it correctly? (seems that no, but why??)

**Reply by shahamran** [reply]
Awesome. I'll try to get to it tonight. Thanks for the responsiveness and pointers!
